src/main/kotlin/com/sourcegraph/cody/chat/ExportChatsBackgroundable.kt

```diff
@@ -0,0 +1,61 @@
+package com.sourcegraph.cody.chat
+
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.protocol.ChatHistoryResponse
+import com.sourcegraph.cody.agent.protocol.IdParam
+import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.cody.history.HistoryService
+import java.util.concurrent.TimeUnit
+
+class ExportChatsBackgroundable(
+    project: Project,
+    private val onSuccess: (ChatHistoryResponse) -> Unit,
+    private val onFinished: () -> Unit
+) : Task.Backgroundable(project, /* title = */ "Exporting chats...", /* canBeCancelled = */ true) {
+
+  override fun run(indicator: ProgressIndicator) {
+    val accountId = CodyAuthenticationManager.instance.getActiveAccount(project)?.id
+    val chats =
+        HistoryService.getInstance(project)
+            .state
+            .chats
+            .filter { it.accountId == accountId }
+            .filter { it.messages.isNotEmpty() }
+
+    chats.forEachIndexed { index, chatState ->
+      AgentChatSessionService.getInstance(project).getOrCreateFromState(chatState)
```

Comment:
TBH I do not like that approach although I know it's easiest from the perspective of the code reuse.
We are creating there a lot more than we need, including all UI panels, fetching LLM models, and what not.

Reply:
What's the purpose of reviving the chat here? `chat/export` below gives you all the chats in local history for the authed account.

Reply:
Some account may not be available in the local history in the agent. We need to restore them from our persistency state so the agent is aware of them.